contracts/CarbonVoteXBasic.sol

```diff
@@ -0,0 +1,183 @@
+        uint votes
+    );
+
+    struct Poll {
+        // voter address => choice => votes
+        // choice can be hashed value of voting options
+        // e.g. sha3("choice #1"), sha3("choice #2")
+        mapping (address => mapping(bytes32 => uint)) votes;
+        // choice => votes
+        mapping (bytes32 => uint) totalVotesByChoice;
+        // voter address => available votes;
+        mapping (address => uint) availableVotes;
+    }
+
+    // pollId => Poll struct
+    mapping (bytes32 => Poll) polls;
+    CarbonVoteXCore public core;
+    bytes32 public namespace;
+
+    /*
+     * Constructor
+     */
+
+    // @param namespace the namespace of this CarbonVoteX receiver
+    // @param coreAddr the address of CarbonVoteXCore
+    constructor (bytes32 _namespace, address coreAddr) public {
+        namespace = _namespace;
+        core = CarbonVoteXCore(coreAddr);
+    }
+
+
+    // @param namespace the namespace of this CarbonVoteX receiver
+    // @param startBlock starting block (unix timestamp) of the event
+    // @param endBlock ending block (unix timestamp) of the event
+    // @param pollId UUID (hash value) of a poll
+    // @param tokenAddr the address of the token
+    // a restricted function
+    // start a new poll
+    // Note that we do not allow re-registrations for the same UUID
+    function startPoll(
+        uint startBlock, 
+        uint endBlock, 
+        bytes32 pollId, 
+        address tokenAddr
+    ) 
+        public
+    {
+        core.register(namespace, startBlock, endBlock, pollId, tokenAddr);
+    }
+
+    // @param pollId UUID (hash value) of a poll
+    // @return the startBlock, endBlock, pollId and token address of the poll.
+    function getPoll (bytes32 pollId) external view returns (uint, uint, bytes32, address) {
+        return core.getPoll(namespace, pollId);
+    }
+
+    // @param pollId UUID (hash value) of a poll
+    // @param voter the address of the voter
+    // @return the amount of gas remining for voter
+    function getGasSent(bytes32 pollId, address voter) external view returns (uint) {
+        return core.getGasSent(namespace, pollId, voter);
+    }
+
+    // @param pollId UUID (hash value) of a poll
+    // @param voter address of a voter
+    // @return the number of available votes
+    function readAvailableVotes(bytes32 pollId, address voter) external view returns (uint) {
+        require(pollExist(pollId));
+        return polls[pollId].availableVotes[voter];
+    }
+
+    // @param pollId UUID (hash value) of a poll
+    // @param choice the choice of votes
+    // @param votes number of votes to redeem for choice
+    // Users can vote with choice "choice" in poll "pollId" with the amount of "votes"
+    function vote(bytes32 pollId, bytes32 choice, uint votes) external {
+        // poll must exits and not yet expired. 
+        require(pollExist(pollId) && !pollExpired(pollId));
+        
+        // voter cannot vote more votes than it has.
```

Comment:
hard to read: add blank line between each commented statement block:
e.g.  
```
        // deduct voter's available votes from.
        polls[pollId].availableVotes[msg.sender] = 
             polls[pollId].availableVotes[msg.sender].sub(votes);

        // place votes to voter's choice. 
       polls[pollId].votes[msg.sender][choice] = 
            polls[pollId].votes[msg.sender][choice].add(votes);

        // place votes to totalVotesByChoice;
        polls[pollId].totalVotesByChoice[choice] = 
             polls[pollId].totalVotesByChoice[choice].add(votes)
```

Reply:
fixed